planet/src/main/webapp/js/itinerary.js

```diff
@@ -224,44 +226,51 @@ function createEventElement(listName, ref, name, address, duration) {
     return eventElement;
 }
 
-function deleteEvent(listName, ref) {
+async function deleteEvent(listName, ref) {
+    const DEFAULT_DATE = '08-07-2020';
     const userId = firebase.auth().currentUser.uid;
     const eventListRef = database.ref('events/' + userId + '/' + listName);
     const toBeDeletedEventRef = eventListRef.child(ref);
+    let listDate = DEFAULT_DATE;
     toBeDeletedEventRef.remove();
 
     // Fix order after deleting the event
     eventListRef.orderByChild('order').once('value', (eventListSnapshot) => {
         let count = 1;
         eventListSnapshot.forEach(function(childEvent) {
-            let event = childEvent.val();
-            let eventKey = childEvent.key;
-            if (event.order !== count) {
-                eventListRef.child(eventKey).update({
-                    order: count
-                });
+            if (childEvent.key !== 'date'){
+                let event = childEvent.val();
+                let eventKey = childEvent.key;
+                if (event.order !== count) {
+                    eventListRef.child(eventKey).update({
+                        order: count
+                    });
+                }
+                count += 1;
+            } else {
+                listDate = childEvent.val();
             }
-            count += 1;
         });
     });
-
-    // Check if this is an event created from a place, and if so, 
-    // remove the user id from the visitors list 
-    const placeRef = database.ref('places/' + ref);
-    placeRef.once('value', (placeSnapshot) => {
-        if (placeSnapshot.val()) {
-            placeRef.child('visitors').child(userId).remove();
-        }
-    });
+    // We check that the list is saved by the user
+    // If the user saved the list then the listDate is not the default value
+    if (listDate !== DEFAULT_DATE) {
```

Comment:
The alternative would be directly checking if listName === 'currentList'

Reply:
I went for idea that at first but then I wasn't sure if you handled the case where the user names the list they want to save "currentList"

Reply:
Good question. I didn't handle that. Maybe we can add an error checking preventing users to name their list as "currentList".

Reply:
Yeah that would be a good idea!